### PR TITLE
Document createClient option

### PR DESCRIPTION
**Pull request by dobesv**
Fix for https://github.com/OptimalBits/bull/issues/1747

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -104,6 +105,19 @@ interface AdvancedSettings {
...
+
+`createClient` is passed a `type` to specify the type of connection that Bull is trying to create, and some options that `bull` would like to set for that connection.
+
+You can merge the provided options with some of your own and create an `ioredis` connection.
+
+When type is `client` or `subscriber` you can return the same connection for multiple queues, which can reduce the number of connections you open to the redis server.  Bull
+does not close or disconnect these connections when queues are closed, so if you need to have your app do a graceful shutdown, you will need to keep references to these
+Redis connections somewhere and disconnect them after you shut down all the queues.
+
+The `bclient` connection however is a "blocking client" and is used to wait for new events on a single queue at a time.  For this reason it cannot be shared and a
```
Comment (manast) [inline REFERENCE.md:118]: new events -> new jobs

**Comment by manast** [comment]
thanks for this great improvement, please check one small comment I made.

File: REFERENCE.md [hunk 2]
```diff
@@ -105,6 +106,19 @@ interface AdvancedSettings {
...
```
Comment (dobesv) [inline REFERENCE.md:119]:
```suggestion
The `bclient` connection however is a "blocking client" and is used to wait for new jobs on a single queue at a time.  For this reason it cannot be shared and a
```

File: REFERENCE.md [hunk 3]
```diff
@@ -105,6 +106,19 @@ interface AdvancedSettings {
...
+`createClient` is passed a `type` to specify the type of connection that Bull is trying to create, and some options that `bull` would like to set for that connection.
+
+You can merge the provided options with some of your own and create an `ioredis` connection.
+
+When type is `client` or `subscriber` you can return the same connection for multiple queues, which can reduce the number of connections you open to the redis server.  Bull
+does not close or disconnect these connections when queues are closed, so if you need to have your app do a graceful shutdown, you will need to keep references to these
+Redis connections somewhere and disconnect them after you shut down all the queues.
+
+The `bclient` connection however is a "blocking client" and is used to wait for new jobs on a single queue at a time.  For this reason it cannot be shared and a
+fresh new connection should be returned each time.  Bull will disconnect this client itself when the queue is closed.
```
Comment (dobesv) [inline REFERENCE.md:120]:
```suggestion
The `bclient` connection however is a "blocking client" and is used to wait for new jobs on a single queue at a time.  For this reason it cannot be shared and a
new connection should be returned each time.
```

File: REFERENCE.md [hunk 4]
```diff
@@ -63,6 +63,7 @@ The optional `url` argument, allows to specify a redis connection string such as
 
 ```typescript
 interface QueueOptions {
+  createClient?(type: 'client' | 'subscriber' | 'bclient', redisOpts?: Redis.RedisOptions): Redis.Redis | Redis.Cluster;
```
Comment (joebowbeer) [inline REFERENCE.md:66]:
```suggestion
  createClient?(type: 'client' | 'subscriber' | 'bclient', config?: Redis.RedisOptions): Redis.Redis | Redis.Cluster;
```

**Pull request merged**